### PR TITLE
[svcr] improve performance of both sample_qc methods

### PR DESCRIPTION
**Pull request by tpoterba**
Also add test for vds.sampleqc.

### CODE REVIEW

**Review by pwc2 — APPROVED** [review]

File: hail/python/test/hail/vds/test_vds.py [hunk 1]
```diff
@@ -38,3 +38,34 @@ def test_conversion_equivalence():
     svcr_readback = hl.vds.to_merged_sparse_mt(vds)
 
     assert svcr._same(svcr_readback)
+
+
+def test_sampleqc_old_new_equivalence():
```
Comment (pwc2) [inline hail/python/test/hail/vds/test_vds.py:43]: I realized after approving that we probably need the failure decorators here too

Comment (tpoterba) [reply hail/python/test/hail/vds/test_vds.py:43]: I think this should work -- the feature that breaks local/service backend above is the VCF combiner. This test just uses plan relational query infrastructure.

Comment (pwc2) [reply hail/python/test/hail/vds/test_vds.py:43]: Gotcha

Comment (tpoterba) [reply hail/python/test/hail/vds/test_vds.py:43]: lint failures though...

**Pull request merged**